crypto/stark-verifier-ethereum/contracts/merkle.sol

```diff
@@ -0,0 +1,122 @@
+pragma solidity ^0.6.4;
+
+import './hashing.sol';
+import './ring_buffer.sol';
+
+contract MerkleVerifier is Hashing, RingBuffer {
+
+    function verify_merkle_proof(
```

Comment:
Please add a line of documentation explaining how the arguments work, how indices are numbered and if they need to be supplied sorted.